### ISSUE TITLE
Enable multi-threading on miners

### ISSUE THREAD

**Issue by obskein** [xgt-network/xgt#13]
Current mining is single threaded, to be more efficient on network/disk/memory, allow multiple mining threads, configured via config.ini

The following code needs to be added to `void witness_plugin::plugin_initialize(const boost::program_options::variables_map& options)` in /libraries/plugins/witness/witness.cpp
```c++
   if( options.count("mining-threads") )
   {
      _mining_threads = std::min( options["mining-threads"].as<uint32_t>(), uint32_t(64) );
      _thread_pool.resize( _mining_threads );
      for( uint32_t i = 0; i < _mining_threads; ++i )
         _thread_pool[i] = std::make_shared<fc::thread>();
   }
```

**Comment by obskein** [comment]
Also of interest is libraries/plugins/witness/witness_plugin.cpp#L262

and the commented out code on

libraries/plugins/witness/witness_plugin.cpp#L313

**Comment by obskein** [comment]
Resolved in #29
